sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -763,6 +776,11 @@ private static bool TryAddExtensionInfo(IDictionary<string, string> extensions,
                         implicitlyRegister = (bool)assemblyAttribute.ConstructorArguments[2].Value;
                     }
 
+                    if (assemblyAttribute.ConstructorArguments.Count == 4)
```

Comment:
Why are we enabling things based on ctor definitions, rather than an actual instance of the assembly attribute? One problem I see with this is that if the extension specifies false for these, e.g. `[assembly: ExtensionInformation("Microsoft.Azure.WebJobs.Extensions.Storage.Blobs", "5.2.0-alpha.20221028.1", false, false)]` you'll still be enabling. Another problem is that you're doing a strict equality check for 4. If someone adds a 5th ctor overload in the future, it'll break your code.

Reply:
For the last point regarding strict equality check, I have updated the check to >=4. And if Extension specifics false then below code will assign supportsDeferredBinding to False

`supportsDeferredBinding = (bool)assemblyAttribute.ConstructorArguments[3].Value;`

Regarding first question, _Why are we enabling things based on ctor definitions, rather than an actual instance of the assembly attribute?_

Need to check with @fabiocav. I remember an old discussion to extend the logic of evaluating constructor arguments.

Reply:
Once you have the attribute instance, why can't you just check attr.SupportsDeferredBinding? Why are you checking ctor params for any of these things that the attribute exposes properties for?

Reply:
I don't think we are getting attribute instance here, instead we are just getting AssemblyDefinition.

```csharp
 private static bool TryAddExtensionInfo(IDictionary<string, string> extensions, AssemblyDefinition extensionAssemblyDefinition, out bool supportsDeferredBinding, bool usedByFunction = true)
        {
            supportsDeferredBinding = false;

            foreach (var assemblyAttribute in extensionAssemblyDefinition.CustomAttributes)
            {
                if (string.Equals(assemblyAttribute.AttributeType.FullName, Constants.ExtensionsInformationType, StringComparison.Ordinal))
                {
                    string extensionName = assemblyAttribute.ConstructorArguments[0].Value.ToString();
                    string extensionVersion = assemblyAttribute.ConstructorArguments[1].Value.ToString();
                    bool implicitlyRegister = false;
                    if (assemblyAttribute.ConstructorArguments.Count >= 3)
                    {
                        // EnableImplicitRegistration
                        implicitlyRegister = (bool)assemblyAttribute.ConstructorArguments[2].Value;
                    }

                    if (assemblyAttribute.ConstructorArguments.Count >= 4)
                    {
                        supportsDeferredBinding = (bool)assemblyAttribute.ConstructorArguments[3].Value;
                    }
```

Reply:
Yeah, it's a little awkward, but this uses `Mono.Cecil`, that isn't working with an actual instance of the attribute in this case.

Reply:
Don't know the backstory on that, but if you're happy, I'm happy